### DISCUSSION TITLE
Add Tracing support for Application Insights/Log Analytics

### DISCUSSION THREAD

**Discussion by stebet** [grafana/grafana#41745] [Data Sources]
**What would you like to be added**:
Azure Application Insights has everything needed to use it as a [data-source for trace views](https://grafana.com/docs/grafana/latest/explore/trace-integration/#data-api). Adding that feature to the Azure Log Analytics data source would be awesome.

```
**Azure Data Type -> Grafana Type**
operation_Id -> traceID
id -> spanID
operation_ParentId -> parentSpanID
appName -> serviceName
customDimensions[] -> serviceTags
timestamp -> startTime
duration -> duration
```

Adding @kylebrandt since I see he has been maintaining a lot of the Azure stuff lately.

**Why is this needed**:
This would enable Grafana trace visualizations for Application Insights telemetry.

**Comment by stebet** [comment]
Example Kusto QL against Application Insights based Azure Log Analytics to return compatible data:
```
AppRequests
| where OperationId == '[SomeAwesomeTraceId]'
| project traceID = OperationId, spanID = Id, parentSpanID = ParentId, serviceName = AppRoleName, serviceTags = bag_merge(pack_dictionary('hostname', AppRoleInstance, 'version', AppVersion), Properties), startTime = TimeGenerated, duration = DurationMs, operationName = OperationName, tags = pack_dictionary('http.url', Url, 'http.status_code', ResultCode, 'error', iff(Success == 'true', 'false', 'true'), 'span.kind', 'server')
| union (AppDependencies
| where OperationId == '[SomeAwesomeTraceId]'
| project traceID = OperationId, spanID = Id, parentSpanID = ParentId, serviceName = AppRoleName, serviceTags = bag_merge(pack_dictionary('hostname', AppRoleInstance, 'version', AppVersion), Properties), startTime = TimeGenerated, duration = DurationMs, operationName = OperationName, tags = case
(
DependencyType == 'HTTP', pack_dictionary('span.kind', 'client', 'peer.hostname', Target, 'error', iff(Success == 'true', 'false', 'true')),
DependencyType == 'Redis', pack_dictionary('span.kind', 'client', 'peer.address', Target, 'db.statement', Data, 'error', iff(Success == 'true', 'false', 'true')),
DependencyType == 'SQL', pack_dictionary('span.kind', 'client', 'peer.address', Target, 'db.statement', Data, 'error', iff(Success == 'true', 'false', 'true')),
dynamic(null))
)
| order by startTime asc
```

It's then easy to also join the results on `AppTraces` and `AppExceptions` to get logs and exception/stack traces as well.

**Comment by jrgcubano** [comment]
Any progress on this?

**Comment by stebet** [comment]
I would love to see this happen since all the dats to do this should be available through Kusto queries.

**Comment by jrgcubano** [comment]
@torkelo @kylebrandt Hello. Any progress on this?

**Comment by jrgcubano** [comment]
It seems to have been implemented in version 10 ([What new in v10](https://grafana.com/docs/grafana/latest/whatsnew/whats-new-in-v10-0/)).

We will have to wait a bit more for Microsoft to update Azure Managed Grafana with version 10.

Thanks for this.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). This looks like it was added to Grafana. If there is still a feature you're missing, feel free to make an [issue](https://github.com/grafana/grafana/issues). Thank you!
